go/types_test.go

```diff
@@ -1,71 +1,52 @@
 )
 
-func TestReadMerkleTreeLeafForX509Cert(t *testing.T) {
-	entry, err := base64.StdEncoding.DecodeString(CertEntryB64)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	m, err := ReadMerkleTreeLeaf(bytes.NewReader(entry))
-	if err != nil {
-		t.Fatal(err)
-	}
-	if m.Version != V1 {
-		t.Fatal("Invalid version number")
-	}
-	if m.LeafType != TimestampedEntryLeafType {
-		t.Fatal("Invalid LeafType")
-	}
-	if m.TimestampedEntry.EntryType != X509LogEntryType {
-		t.Fatal("Incorrect EntryType")
-	}
-}
-
-func TestReadMerkleTreeLeafForPrecert(t *testing.T) {
-	entry, err := base64.StdEncoding.DecodeString(PrecertEntryB64)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	m, err := ReadMerkleTreeLeaf(bytes.NewReader(entry))
-	if err != nil {
-		t.Fatal(err)
-	}
-	if m.Version != V1 {
-		t.Fatal("Invalid version number")
-	}
-	if m.LeafType != TimestampedEntryLeafType {
-		t.Fatal("Invalid LeafType")
-	}
-	if m.TimestampedEntry.EntryType != PrecertLogEntryType {
-		t.Fatal("Incorrect EntryType")
-	}
-}
-
-func TestReadMerkleTreeLeafChecksVersion(t *testing.T) {
-	buffer := []byte{1}
-	_, err := ReadMerkleTreeLeaf(bytes.NewReader(buffer))
-	if err == nil || !strings.Contains(err.Error(), "unknown Version") {
-		t.Fatal("Failed to check Version - accepted 1")
-	}
-}
-
-func TestReadMerkleTreeLeafChecksLeafType(t *testing.T) {
-	buffer := []byte{0, 0x12, 0x34}
-	_, err := ReadMerkleTreeLeaf(bytes.NewReader(buffer))
-	if err == nil || !strings.Contains(err.Error(), "unknown LeafType") {
-		t.Fatal("Failed to check LeafType - accepted 0x1234")
+func TestReadMerkleTreeLeaf(t *testing.T) {
```

Comment:
It does just check that the `tls` library does what it's supposed to, but there's no harm in keeping it.  Renamed.